src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle'| 'circle' | 'rectangle';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number,
+}
+
+function isGoodTriangle(...sides:number[]):boolean {
+  const copySides = [...sides];
+  const max = Math.max(...copySides);
+
+  copySides.splice(copySides.indexOf(max));
 
+  return (copySides[0] + copySides[1]) > max;
+}
+
+function toFixedArea(area:number):number {
+  const fixed = 10 ** 2;
+
+  return Math.floor(area * fixed) / fixed;
 }
 
 export class Triangle {
+  public readonly shape:Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || !isGoodTriangle(this.a, this.b, this.c)) {
```

Comment:
you can check it only by `if (a + b <= c || a + c <= b || c + b <= a)` without creating additional function (you use this function in code only once)

Reply:
Please - don't ignore the comment - either fix the issue or comment on why you make other decision